### ISSUE TITLE
[cloud_firestore] Accessing User Image with Future Builder

### ISSUE THREAD

**Issue by SportApps** [FirebaseExtended/flutterfire#2620]
Hi,

I am trying to load my user Image on a start screen. It must be rebuild always when I enter the screen, as there is a function in the app that allows to change the User Image.

This is why I decided to use the Future Builder here.

The updated Image Url is stored in Firestore - collection (users) - document (userurl) - "image_url" field.

THE PROBLEM
The image actually loads correctly - but after aprox 5 sec. Before that there appear several errors:

`java.lang.IllegalArgumentException: Invalid document reference. Document references must have an even number of segments, but users has 1
`
I suspect it is because the user Uid must first be loaded and the build message starts for some reason before the future is completed...

Can somebody help? Would be very much appreciated.

CODE:

I created an async updateImage Function

```dart
FirebaseAuth auth = FirebaseAuth.instance;
  Future setImageLink() async {
    var myuserdata = await auth.currentUser();
    _userUid = await myuserdata.uid;
    var firestoreInstance =
        await Firestore.instance.collection("users").document(_userUid).get();
    return firestoreInstance;
  }
```

The Future Builder is in my Scaffold widget:

```dart
    body: Column(
        children: <Widget>[
          Column(
            children: <Widget>[
              FutureBuilder(
                  future: setImageLink(),
                  builder: (context, usnapshot) {
                    print(usnapshot.data["image_url"]);
                    if (usnapshot.connectionState == ConnectionState.none &&
                        usnapshot.data == null) {
                      print('project snapshot data is: ${usnapshot.data}');
                      return Container();
                    }
                    if (usnapshot.hasError){return Text("One Moment");}
                    if (usnapshot.hasData && usnapshot.connectionState == ConnectionState.done) {
                      return UserInfoStartscreen(
                        name: _username,
                        skill: _level,
                        heightContainer: _heightUserC,
                        imageUrl: usnapshot.data["image_url"],
                        userUrl: _userUid,
                      );
                    }
                  }),
            ],
          ),
```

```console
C:\flutter\bin\flutter.bat doctor --verbose
[√] Flutter (Channel stable, v1.17.1, on Microsoft Windows [Version 10.0.18362.836], locale en-US)
    • Flutter version 1.17.1 at C:\flutter
    • Framework revision f7a6a7906b (13 days ago), 2020-05-12 18:39:00 -0700
    • Engine revision 6bc433c6b6
    • Dart version 2.8.2

[!] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at C:\Users\Usuario\AppData\Local\Android\sdk
    • Platform android-28, build-tools 28.0.3
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b04)
    ! Some Android licenses not accepted.  To resolve this, run: flutter doctor --android-licenses

[√] Android Studio (version 3.6)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 45.1.1
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b04)

[!] IntelliJ IDEA Community Edition (version 2018.1)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2018.1.4
    X Flutter plugin not installed; this adds Flutter specific functionality.
    X Dart plugin not installed; this adds Dart specific functionality.
    • For information about installing plugins, see
      https://flutter.dev/intellij-setup/#installing-the-plugins

[√] Connected device (1 available)
    • sdk gphone x86 • emulator-5554 • android-x86 • Android 10 (API 29) (emulator)

! Doctor found issues in 2 categories.
Process finished with exit code 0
```

```yaml
name: tournament2020
description: A new Flutter application.

# The following line prevents the package from being accidentally published to
# pub.dev using `pub publish`. This is preferred for private packages.
publish_to: 'none' # Remove this line if you wish to publish to pub.dev

# The following defines the version and build number for your application.
# A version number is three numbers separated by dots, like 1.2.43
# followed by an optional build number separated by a +.
# Both the version and the builder number may be overridden in flutter
# build by specifying --build-name and --build-number, respectively.
# In Android, build-name is used as versionName while build-number used as versionCode.
# Read more about Android versioning at https://developer.android.com/studio/publish/versioning
# In iOS, build-name is used as CFBundleShortVersionString while build-number used as CFBundleVersion.
# Read more about iOS versioning at
# https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/CoreFoundationKeys.html
version: 1.0.0+1

environment:
  sdk: ">=2.7.0 <3.0.0"

dependencies:
  flutter:
    sdk: flutter


  # The following adds the Cupertino Icons font to your application.
  # Use with the CupertinoIcons class for iOS style icons.
  cupertino_icons: ^0.1.3
  cloud_firestore: ^0.13.4
  firebase_auth: ^0.16.0
  firebase_storage: ^3.1.5
  image_picker: ^0.6.6+1
  firebase_messaging: ^6.0.15
  hexcolor: ^1.0.2

dev_dependencies:
  flutter_test:
    sdk: flutter

# For information on the generic Dart part of this file, see the
# following page: https://dart.dev/tools/pub/pubspec

# The following section is specific to Flutter.
flutter:

  # The following line ensures that the Material Icons font is
  # included with your application, so that you can use the icons in
  # the material Icons class.
  uses-material-design: true

  # To add assets to your application, add an assets section, like this:
  # assets:
  #   - images/a_dot_burr.jpeg
  #   - images/a_dot_ham.jpeg

  # An image asset can refer to one or more resolution-specific "variants", see
  # https://flutter.dev/assets-and-images/#resolution-aware.

  # For details regarding adding assets from package dependencies, see
  # https://flutter.dev/assets-and-images/#from-packages

  # To add custom fonts to your application, add a fonts section here,
  # in this "flutter" section. Each entry in this list should have a
  # "family" key with the font family name, and a "fonts" key with a
  # list giving the asset and other descriptors for the font. For
  # example:
  # fonts:
  #   - family: Schyler
  #     fonts:
  #       - asset: fonts/Schyler-Regular.ttf
  #       - asset: fonts/Schyler-Italic.ttf
  #         style: italic
  #   - family: Trajan Pro
  #     fonts:
  #       - asset: fonts/TrajanPro.ttf
  #       - asset: fonts/TrajanPro_Bold.ttf
  #         weight: 700
  #
  # For details regarding fonts from package dependencies,
  # see https://flutter.dev/custom-fonts/#from-packages
```

**Comment by iapicca** [comment]
Hi @SportApps 
can you please provide your `flutter doctor -v` , 
your `flutter run --verbose`
and your `pubspec.yaml` 
Thank you

**Comment by SportApps** [comment]
> Hi @SportApps
> can you please provide your `flutter doctor -v` ,
> your `flutter run --verbose`
> and your `pubspec.yaml`
> Thank you



> Hi @SportApps
> can you please provide your `flutter doctor -v` ,
> your `flutter run --verbose`
> and your `pubspec.yaml`
> Thank you

Ok, I just edited my initial post with the info.

**Comment by TahaTesser** [comment]
Hi @SportApps 
Can you please provide your a minimal complete reproducible code sample, your code sample has mossing code 
Thank you

**Comment by Salakar** [comment]
Hey, as part of our on-going work for #2582, this has been resolved in our Firebase Firestore rework (#2913) - which has now been merged into master. We'll look at publishing some prereleases in the next few days. Thank you
